baseplate/events/publisher.py

```diff
@@ -4,34 +4,40 @@
 import hashlib
 import hmac
 import logging
-import time
 
 import requests
 
 from . import MAX_EVENT_SIZE, MAX_QUEUE_SIZE
 from .. import config, metrics_client_from_config
 from .. _compat import configparser, BytesIO
-from .. message_queue import MessageQueue, TimedOutError
 from .. _utils import (
     Batch, BatchFull, RawJSONBatch,
     SerializedBatch, TimeLimitedBatch,
 )
+from .. message_queue import MessageQueue, TimedOutError
+from .. retry import RetryPolicy
 
 
 logger = logging.getLogger(__name__)
 
 
 # seconds to wait for the event collector
 POST_TIMEOUT = 3
-# seconds to wait between retries when the collector fails
-RETRY_DELAY = 2
+# Base rate for expontential retry delay
+RETRY_BACKOFF = 2
+# Maximum time spent wait for a message to be sent
+MAX_RETRY_TIME = 15 * 60
```

Comment:
that's quite a while! do we want to bail out and let init restart us sooner or maybe lessen the base rate of backoff?

after only a few tries we're waiting minutes between tries which seems pretty intense for recovery.